### PR TITLE
New block - get sprites within

### PR DESCRIPTION
**Pull request by livcheerful**
image

I need to update the category/order >> can put it anywhere

### CODE REVIEW

File: spriteutils.ts [hunk 1]
```diff
@@ -39,6 +39,39 @@ namespace spriteutils {
...
+            if (distanceA < distanceB) {
+                return -1;
+            } else if (distanceA > distanceB) {
+                return 1;
+            } else {
+                return 0;
+            }
+        })
+
+        return sortArray? sortArray:[];
```
Comment (jwunderl) [inline spriteutils.ts]:
```suggestion
        return sortArray ? sortArray : [];
```

File: spriteutils.ts [hunk 2]
```diff
@@ -39,6 +39,39 @@ namespace spriteutils {
...
+     * Returns array of sprites of kind that are within a specified distance.
+     */
+    //% block="get all sprites of kind $kind within $distance pixels from $sprite"
+    //% kind.shadow=spritekind
+    //% distance.defl=50
+    //% sprite.shadow=variables_get
+    //% sprite.defl=mySprite
+    export function getSpritesWithin(kind: number, distance: number, sprite: Sprite) : Sprite[] {
+        let allSprites = sprites.allOfKind(kind);
+        let sortArray : Sprite[] = [];
```
Comment (jwunderl) [inline spriteutils.ts]:
```suggestion
        let sortArray: Sprite[] = [];
```

File: spriteutils.ts [hunk 3]
```diff
@@ -39,6 +39,39 @@ namespace spriteutils {
...
 
+    /**
+     * Returns array of sprites of kind that are within a specified distance.
+     */
+    //% block="get all sprites of kind $kind within $distance pixels from $sprite"
+    //% kind.shadow=spritekind
+    //% distance.defl=50
+    //% sprite.shadow=variables_get
+    //% sprite.defl=mySprite
+    export function getSpritesWithin(kind: number, distance: number, sprite: Sprite) : Sprite[] {
```
Comment (jwunderl) [inline spriteutils.ts]:
```suggestion
    export function getSpritesWithin(kind: number, distance: number, sprite: Sprite): Sprite[] {
```

File: spriteutils.ts [hunk 4]
```diff
@@ -39,6 +39,39 @@ namespace spriteutils {
...
+        let allSprites = sprites.allOfKind(kind);
+        let sortArray : Sprite[] = [];
+        let numItems = 0;
+        allSprites.forEach(function(value: Sprite, index: number) {
```
Comment (jwunderl) [inline spriteutils.ts]: how about `sortArray = allSprites.filter(...)` here?

File: spriteutils.ts [hunk 5]
```diff
@@ -39,6 +39,39 @@ namespace spriteutils {
...
+            if (distanceBetween(value, sprite) <= distance) {
+                sortArray.push(value);
+            }
+        })
+
+        sortArray = sortArray.sort((a, b)=>{
+            let distanceA = (distanceBetween(a, sprite));
```
Comment (jwunderl) [inline spriteutils.ts]:
```suggestion
            let distanceA = distanceBetween(a, sprite);
```

File: spriteutils.ts [hunk 6]
```diff
@@ -39,6 +39,39 @@ namespace spriteutils {
...
+            let distanceB = distanceBetween(b, sprite)
+            if (distanceA < distanceB) {
```
Comment (jwunderl) [inline spriteutils.ts:63]: can just return distanceA - distanceB here right?

**Review by jwunderl — COMMENTED** [review]
few nits :P

also need a docs page but otherwise looks good :)

example of adding there here: https://github.com/jwunderl/arcade-sprite-util/commit/568ed8c20c47927eda5b92c5adc861add387ccf6 you just do the `+ new file` & `docs/get-sprites-within` and it will add the docs correctly

File: spriteutils.ts [hunk 7]
```diff
@@ -39,6 +39,39 @@ namespace spriteutils {
...
+                return -1;
+            } else if (distanceA > distanceB) {
+                return 1;
+            } else {
+                return 0;
+            }
+        })
```
Comment (jwunderl) [inline spriteutils.ts]:
```suggestion
        sortArray = sortArray.sort((a, b) => distanceBetween(a, sprite) - distanceBetween(b, sprite));
```

can just return distanceA - distanceB here right?

File: spriteutils.ts [hunk 8]
```diff
@@ -39,6 +39,31 @@ namespace spriteutils {
...
+    //% distance.defl=50
+    //% sprite.shadow=variables_get
+    //% sprite.defl=mySprite
+    //% weight=85
+    //% group=Sprite
+    export function getSpritesWithin(kind: number, distance: number, sprite: Sprite): Sprite[] {
+        let allSprites = sprites.allOfKind(kind);
+        let sortArray: Sprite[] = [];
+        let numItems = 0;
+        sortArray = allSprites.filter(function(value: Sprite, index: number) {
```
Comment (jwunderl) [inline spriteutils.ts]:
```suggestion
        let numItems = 0;
        let sortArray = allSprites.filter(function(value: Sprite, index: number) {
```

File: spriteutils.ts [hunk 9]
```diff
@@ -39,6 +39,31 @@ namespace spriteutils {
...
+            return distanceBetween(value, sprite) <= distance;
+        })
+
+        sortArray = sortArray.sort((a, b)=>{
+            return (distanceBetween(a, sprite) - distanceBetween(b, sprite));
+        })
+
+        return sortArray ? sortArray : [];
```
Comment (jwunderl) [inline spriteutils.ts]:
```suggestion
        return sortArray;
```

**Review by jwunderl — APPROVED** [review]
two more minor nits but LGTM~

**Comment by jwunderl** [comment]
(@livcheerful you should press the green button!)

**Comment by jwunderl** [comment]
Nice nice, thankyou!

**Comment by livcheerful** [comment]
yay! I learned :D

**Comment by jwunderl** [comment]
Last part is to go back to the project in arcade, press the 'pull' button to get back up to date with master, and then bump a new release (minor is fine)!

**Pull request merged**